cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/analyzer/goals/RackAwareGoal.java

```diff
@@ -71,11 +85,23 @@ public boolean isActionAcceptable(BalancingAction action, ClusterModel clusterMo
       // Remove brokers in partition broker racks except the brokers in replica broker rack.
       for (Broker broker : partitionBrokers) {
         if (broker.rack().brokers().contains(destinationBroker)) {
-          return false;
+          return BROKER_REJECT;
+        }
+      }
+      if (action.balancingAction() == ActionType.REPLICA_SWAP) {
+        // Destination broker cannot be in a rack that violates rack awareness.
```

Comment:
The logic here is a little confusing. If source and destination broker are in the same rack, we should allow swap instead of returning REPLICA_REJECT, right?

Reply:
Simplified the logic.